index.js

```diff
@@ -235,6 +235,12 @@ function addPreflightHeaders (req, reply, corsOptions) {
   if (corsOptions.maxAge !== null) {
     reply.header('Access-Control-Max-Age', String(corsOptions.maxAge))
   }
+
+  if (typeof corsOptions.cacheControl === 'number') {
```

Comment:
Do you think this would be a better validation?

```suggestion
  if (Number.isInteger(Number(corsOptions.cacheControl)) === true) {
```

String numbers will be coerced, and all values will be verified to be integers (as per spec).

Reply:
How about patching normalizeCorsOptions?